supertokens_python/session/session_recipe.py

```diff
@@ -0,0 +1,247 @@
+                       self.config.sign_out_feature.disable_default_implementation)
+        ]
+
+    async def handle_api_request(self, request_id: str, request: BaseRequest, _: NormalisedURLPath, __: str):
+        if request_id == SESSION_REFRESH:
+            return await handle_refresh_api(self, request)
+        else:
+            return await handle_signout_api(self, request)
+
+    async def handle_error(self, request: BaseRequest, error: SuperTokensError, response : BaseResponse):
+        if isinstance(error, UnauthorisedError):
+            return await self.config.error_handlers.on_unauthorised(request, str(error), response)
+        elif isinstance(error, TokenTheftError):
+            return await self.config.error_handlers.on_token_theft_detected(request, error.session_handle, error.user_id, response)
+        else:
+            return await self.config.error_handlers.on_try_refresh_token(request, str(error), response)
+
+    def get_all_cors_headers(self) -> List[str]:
+        return get_cors_allowed_headers()
+
+    @staticmethod
+    def init(config=None):
+        def func(app_info: AppInfo):
+            if SessionRecipe.__instance is None:
+                SessionRecipe.__instance = SessionRecipe(SessionRecipe.recipe_id, app_info, config)
+                return SessionRecipe.__instance
+            else:
+                raise_general_exception(None, 'Session recipe has already been initialised. Please check your code for bugs.')
+        return func
+
+    @staticmethod
+    def get_instance() -> SessionRecipe:
+        if SessionRecipe.__instance is not None:
+            return SessionRecipe.__instance
+        raise_general_exception(None, 'Initialisation not done. Did you forget to call the SuperTokens.init function?')
+
+    @staticmethod
+    def reset():
+        if ('SUPERTOKENS_ENV' not in environ) or (
+                environ['SUPERTOKENS_ENV'] != 'testing'):
+            raise_general_exception(None, 'calling testing function in non testing env')
+        SessionRecipe.__instance = None
+
+    # instance functions below...............
+
+    async def get_handshake_info(self) -> HandshakeInfo:
+        if self.handshake_info is None:
+            ProcessState.get_instance().add_state(AllowedProcessStates.CALLING_SERVICE_IN_GET_HANDSHAKE_INFO)
+            response = await self.get_querier().send_post_request(NormalisedURLPath(self, '/recipe/handshake'), {})
+            self.handshake_info = HandshakeInfo({
+                **response,
+                'antiCsrf': self.config.anti_csrf
+            })
+        return self.handshake_info
+
+    def update_jwt_signing_public_key_info(self, new_key, new_expiry):
+        if self.handshake_info is not None:
+            self.handshake_info.update_jwt_signing_public_key_info(new_key, new_expiry)
+
+    async def create_new_session(self, request: BaseRequest, user_id: str, jwt_payload: Union[dict, None] = None,
```

Comment:
this is a use facing function which means it can be directly used by the user in their API handler. We shouldn't expect them to wrap the request based onto their framework. Rather check here if the request is wrapped or not. if it is not wrapped, then wrap it based on the framework config